source/extensions/geoip_providers/maxmind/geoip_provider.cc

```diff
@@ -165,7 +165,10 @@ void GeoipProvider::lookupInCityDb(
     auto city_db_ptr = getCityDb();
     // Used for testing.
     synchronizer_.syncPoint(std::string(CITY_DB_TYPE).append("_lookup_pre_complete"));
-    RELEASE_ASSERT(city_db_ptr, "Maxmind city database must be initialised for performing lookups");
+    if (!city_db_ptr) {
+      IS_ENVOY_BUG("Maxmind city database must be initialised for performing lookups");
```

Comment:
After more thinking, switched from crashing semantics with `RELEASE_ASSERT` to `IS_ENVOY_BUG`, as entire process does not need to die if geolocation db is nullptr (except of initial load). It will log error, inc error stat and skip geolocation header decoration for that db type in release mode.